docs/markdown/Reference-manual.md

```diff
@@ -741,6 +743,7 @@ This object is returned by [`meson.get_compiler(lang)`](#meson-object). It repre
 
 - `get_id()` returns a string identifying the compiler (e.g. `'gcc'`)
 - `version()` returns the compiler's version number as a string
+- `cmd_array()` returns the command to invoke the compiler as a list (e.g. `['ccache', 'cc']`), available since 0.24.0
```

Comment:
This function was added for use in tests, and is not documented on purpose. It is our concern that people will look at the compiler object, find this, and start using it to compile everything instead of using `executable()` or `library()`, etc. That would be a terrible situation because it means that your compiled objects won't integrate with the project options, default compiler args, and so on.

If we do document this, I think we should add some text that makes it clear that this is supposed to be used for edge-cases, and that people should report their edge-cases to us so we can figure out a way to support it without them having to resort to using this.